### PR TITLE
added usecase architecture

### PR DESCRIPTION
**Pull request by monteiro**
Issue #8

### CODE REVIEW

File: Request/GetInboxMessagesRequestBuilder.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public function setMaxResults($maxResults)
+    {
+        $this->dto->maxResults = $maxResults;
+
+        return $this;
+    }
+
+    public function build()
```
Comment (xipasduarte) [inline Request/GetInboxMessagesRequestBuilder.php]: Where is the "building"? This seams to just return something unchanged as in a `getDto()`

File: Tests/Request/GetInboxMessagesRequestBuilderTest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        $this->builder = new GetInboxMessagesRequestBuilder();
+    }
+
+    /**
+     * Tests the builder creating all the parameters available to set and verify them
+     */
+    public function testBuilder()
+    {
+        $maxResults = 100;
```
Comment (xipasduarte) [inline Tests/Request/GetInboxMessagesRequestBuilderTest.php]: Use as `const MAX_RESULTS = 100;` like in `Tests/Usecase/Inbox/GetInboxMessagesUsecaseTest.php`

File: Request/GetInboxMessagesRequestBuilder.php (hunk 1 above)
Comment (monteiro) [inline Request/GetInboxMessagesRequestBuilder.php]: True, the goal here was to make some sort of order like:
first create, then you set the parameters, then you build to return what you have just created.

File: Tests/Usecase/Inbox/GetInboxMessagesUsecaseTest.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        $response = $this->usecase->execute($request);
+
+        $this->assertInstanceOf('Symfony\Component\HttpFoundation\JsonResponse', $response);
+    }
+
+    private function getRequest()
+    {
+        $requestBuilder = new GetInboxMessagesRequestBuilder();
+        $request = $requestBuilder->create()->setMaxResults(GetInboxMessagesUsecaseTest::MAX_RESULTS)->build();
```
Comment (xipasduarte) [inline Tests/Usecase/Inbox/GetInboxMessagesUsecaseTest.php]: `setMaxResults()` returns the same as `build()`

Comment (monteiro) [inline Tests/Usecase/Inbox/GetInboxMessagesUsecaseTest.php]: ```setMaxResults()``` return a requestBuilder. 
```build``` returns the DTO.

Comment (xipasduarte) [inline Tests/Usecase/Inbox/GetInboxMessagesUsecaseTest.php]: Yup, my bad!

**Pull request merged**